src/txn/txn_recover.c

```diff
@@ -452,6 +452,68 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the checkpoint snapshot details as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG list;
+    WT_CONFIG_ITEM cval;
+    WT_CONFIG_ITEM k, v;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    uint8_t counter;
+    char *endptr;
+    char *sys_config;
+
+    sys_config = NULL;
+    session = r->session;
+    conn = S2C(session);
+    counter = 0;
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MIN, &cval));
+        if (cval.len != 0)
+            conn->txn_global.snapshot_min = (uint16_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MAX, &cval));
+        if (cval.len != 0)
+            conn->txn_global.snapshot_max = (uint16_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_COUNT, &cval));
+        if (cval.len != 0)
+            conn->txn_global.snapshot_count = (uint16_t)cval.val;
+
+        WT_ERR(__wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT, &cval));
+        if (cval.len != 0) {
+            __wt_config_subinit(session, &list, &cval);
+            WT_ERR(__wt_calloc_def(
+              session, conn->txn_global.snapshot_count, &conn->txn_global.snapshots));
+            while (__wt_config_next(&list, &k, &v) == 0) {
+                if (k.type == WT_CONFIG_ITEM_NUM)
+                    conn->txn_global.snapshots[counter++] = (uint64_t)strtoll(k.str, &endptr, 10);
```

Comment:
Thanks @agorrod for pointing this.  The reason I used `strtoll` because `__wt_config_next(&list, &k, &v)` [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/config/config.c#L510) just processes the value of `v` and does not process the value of `k`. So, I have to explicitly check the type of `k` and call `strtoll` on k.str to convert `k.str` to an integer.

Reply:
Another alternative would be to add a function say `__wt_process_config_next(WT_CONFIG *conf, WT_CONFIG_ITEM *key, WT_CONFIG_ITEM *value)` which would process both key and value part. So in this case, the calling function does not have to invoke `strtoll` and use `k.val` instead.

Something like below, I tried and it works. 

```
int
__wt_process_config_next(WT_CONFIG *conf, WT_CONFIG_ITEM *key, WT_CONFIG_ITEM *value)
{
    WT_RET(__config_next(conf, key, value));
    __config_process_value(key);
    __config_process_value(value);
    return (0);
}
```

Reply:
I see. It seems like there is something missing from the config parsing code that we need here to extract a list of un-named things. There is __wt_config_subgets to look for a string in a list. I think we want a __wt_config_subget_next(WT_CONFIG *conf, WT_CONFIG_ITEM *key);
If I understand correctly - there isn't actually a value here, it's just the key we want.

Reply:
Yes, in our scenario it is just the `key` and we need `key.val` to be set. If I understand correctly, I have added something like below (reusing `__config_next`). 

```
int
__wt_config_subget_next(WT_CONFIG *conf, WT_CONFIG_ITEM *key)
{
    WT_CONFIG_ITEM value;
    WT_RET(__config_next(conf, key, &value));
    __config_process_value(key);
    return (0);
}
```